webui/react/src/pages/JobQueue/JobQueue.tsx

```diff
@@ -0,0 +1,323 @@
+    if (job.type === Api.Determinedjobv1Type.EXPERIMENT) {
+      triggers['Kill'] = async () => {
+        await killExperiment({ experimentId: parseInt(job.entityId, 10) });
+      };
+    }
+
+    return triggers;
+  }, []);
+
+  const hideModal = useCallback(() => setManagingJob(undefined), []);
+
+  const fakeTaskFromJob = (job: Job): AnyTask => {
+    const baseTask = { id: job.entityId, name: job.name };
+    let rv: AnyTask;
+    if (job.type === Api.Determinedjobv1Type.EXPERIMENT) {
+      rv = { ...baseTask, archived: false, state: RunState.Active } as ExperimentTask;
+    } else {
+      rv = {
+        ...baseTask,
+        state: CommandState.Running,
+        type: determinedjobv1TypeToCommandType(job.type),
+      } as CommandTask;
+    }
+    return rv;
+  };
+
+  const columns = useMemo(() => {
+    return defaultColumns.map(col => {
+      switch (col.key) {
+        case 'actions':
+          col.render = (_, record) => {
+            return <div>
+              <ActionDropdown<JobAction>
+                actionOrder={[
+                  'Manage Job',
+                  'Move to Top',
+                  'Cancel',
+                  'Kill',
+                ]}
+                confirmations={{
+                  'Cancel': { cancelText: 'Abort' },
+                  'Kill': {},
+                  'Move to Top': {},
+                }}
+                thing={{ id: record.name, kind: 'job' }}
+                onTrigger={dropDownOnTrigger(record)}
+              />
+              <TaskActionDropdown task={fakeTaskFromJob(record)} />
+            </div>;
+          };
+          break;
+        case SCHEDULING_VAL_KEY:
+          switch (selectedRp?.schedulerType) {
+            case Api.V1SchedulerType.PRIORITY:
+              col.title = 'Priority';
+              col.dataIndex = 'priority';
+              break;
+            case Api.V1SchedulerType.FAIRSHARE:
+              col.title = 'Weight';
+              col.dataIndex = 'weight';
+              break;
+            case Api.V1SchedulerType.KUBERNETES: // FIXME
+              col.title = 'K8';
+              col.dataIndex = 'priority';
+              break;
+          }
+          break;
+        case 'jobsAhead':
+          if (selectedRp && selectedRp.schedulerType !== Api.V1SchedulerType.PRIORITY) {
+            col.render = (_: unknown, record) => {
+              return <div>
+                {!record.isPreemptible && <Icon name="lock" />}
+              </div>;
+            };
+            col.sorter = undefined;
+            col.title = 'Preemptible';
+          } else {
+            col.render = (_: unknown, record) => {
+              return <div>
+                {record.summary.jobsAhead}
+                {!record.isPreemptible && <Icon name="lock" />}
+              </div>;
+            };
+            col.sorter = (a: Job, b: Job): number =>
+              numericSorter(a.summary.jobsAhead, b.summary.jobsAhead);
```

Comment:
ditto

Reply:
ah it wasn't optional when I initialy wrote this. https://github.com/determined-ai/determined/pull/3167#discussion_r749773948 I'll take care of it.